spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java

```diff
@@ -145,14 +146,43 @@ public class TaskTemplate implements TaskOperations {
 
 		this.restTemplate = restTemplate;
 		this.dataFlowServerVersion = dataFlowServerVersion;
+		String version = VersionUtils.getThreePartVersion(dataFlowServerVersion);
+		if (VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(version, EXECUTIONS_CURRENT_RELATION_VERSION)) {
+			Assert.isTrue(resources.getLink(EXECUTIONS_CURRENT_RELATION).isPresent(), ()-> EXECUTIONS_CURRENT_RELATION + " relation is required");
+			this.executionsCurrentLink = resources.getLink(EXECUTIONS_CURRENT_RELATION).get();
+		} else {
+			this.executionsCurrentLink = null;
+		}
+		if (VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(version, VALIDATION_RELATION_VERSION)) {
+			Assert.isTrue(resources.getLink(VALIDATION_REL).isPresent(), ()->VALIDATION_REL + " relation is required");
+			this.validationLink = resources.getLink(VALIDATION_REL).get();
+		} else {
+			this.validationLink = null;
+		}
 
-		this.validationLink = resources.getLink(VALIDATION_REL).get();
```

Comment:
Why these drastic changes in this subsequent review commit? I'm just not sure what this is in response to?